app/javascript/packages/captcha-submit-button/captcha-submit-button-element.ts

```diff
@@ -1,8 +1,16 @@
 export const CAPTCHA_EVENT_NAME = 'lg:captcha-submit-button:challenge';
 
 class CaptchaSubmitButtonElement extends HTMLElement {
+  form: HTMLFormElement | null;
+
   connectedCallback() {
-    this.button.addEventListener('click', (event) => this.handleButtonClick(event));
+    this.form = this.closest('form');
+
+    this.form?.addEventListener('submit', this.handleFormSubmit);
```

Comment:
This dependency on an external element when the node is connected is a bit clunky to deal with, but unfortunately the form's `submit` event is the most reliable way to know that all of the client-side validation has passed.

Reply:
It's an edge-case, but I was curious the specific behavior of custom elements when being moved around the page while already live / attached. It looks like they will call `disconnectedCallback` and then `connectedCallback` in its new position, which will behave how we want it to.

https://codepen.io/aduth/pen/rNZodWj?editors=1111